stock_location_orderpoint/models/stock_location_orderpoint.py

```diff
@@ -0,0 +1,412 @@
+            "location_orderpoint_id": self.id,
+        }
+
+    def _get_waiting_move_domain(self):
+        """
+        Returns a domain which selects waiting moves
+        which should be replenished by given orderpoints
+        """
+        domain = [
+            ("state", "in", ["confirmed", "partially_available"]),
+            ("move_orig_ids", "=", False),
+            ("procure_method", "=", "make_to_stock"),
+        ]
+        location_domains = []
+        for orderpoint in self:
+            location_domains.append(
+                [
+                    ("location_id", "child_of", orderpoint.location_id.ids),
+                    "!",
+                    ("location_dest_id", "child_of", orderpoint.location_id.ids),
+                ]
+            )
+        if location_domains:
+            domain = expression.AND([domain, expression.OR(location_domains)])
+        return domain
+
+    def _find_potential_moves_to_replenish_by_location(self, products=False):
+        """Return a dictionary of products per location that potentially require a replenishment
+        based on the fact there are moves not reserved for those products.
+        This reduces the list of products for which the quantity will be computed"""
+        domain = self._get_waiting_move_domain()
+        if products:
+            domain = expression.AND([domain, [("product_id", "in", products.ids)]])
+        moves_grouped = self.env["stock.move"].read_group(
+            domain,
+            ["ids:array_agg(id)", "location_id"],
+            "location_id",
+        )
+        return {
+            self.env["stock.location"]
+            .browse(res["location_id"][0]): self.env["stock.move"]
+            .browse(res["ids"])
+            for res in moves_grouped
+        }
+
+    def _sort_orderpoints(self):
+        return self.sorted()
+
+    @api.model
+    def _compute_quantities_dict(self, locations, products):
+        qties = {}
+        for location in locations:
+            qties_on_location = qties.setdefault(location, {})
+            products = products.with_context(location=location.id)
+            for product_id, qties_dict in products._product_available().items():
+                product = products.browse(product_id)
+                qties_on_location[product] = qties_dict
+        return qties
+
+    def _get_qty_to_replenish(
+        self, product, qties_on_locations, qty_already_replenished=0
+    ):
+        """
+        Returns a qty to replenish for a given orderpoint and product
+        Calls the _get_qty_to_replenish_{self.replenish_method}
+        To add a new rule how the qty_to_replenish is computed
+        simple add a new
+         - value in replenish_method selection
+         - method called _get_qty_to_replenish_<your_new_replenish_method>
+        """
+        self.ensure_one()
+        product.ensure_one()
+
+        qty_to_replinsh_func_name = f"_get_qty_to_replenish_{self.replenish_method}"
```

Comment:
I'm not a fan of this kind of extension mechanism relaying on code introspection. It make code difficult to debug since it's not possible to easily find where a method is used (your ide  or the grep commande will never find where the `_get_qty_to_replenish_fill_up` method is used) Explicit is always better than implicit ant the code will also be easier to read....
```python
def _get_qty_to_replenish(
        self, product, qties_on_locations, qty_already_replenished=0
    ):
  if self.replenish_method == "fill_up":
    return self._get_qty_to_replenish_fill_up(product, qties_on_locations, qty_already_replenished)
  return 0
´´´

Reply:
Thats fine for me,